src/helpers/connectorConfig/FormField.tsx

```
@@ -0,0 +1,456 @@
+  }
+
+  const fieldPath = field.path.join('.');
+
+  // For child fields (oneOf subfields), watch the parent field to get the current selected value
+  let parentValue = propParentValue;
+  if (field.parentValue !== undefined && !propParentValue) {
+    // Find the parent field path by removing the last segment
+    const parentFieldPath = field.path.slice(0, -1).join('.');
+    if (parentFieldPath) {
+      // Watch only the specific parent field to avoid unnecessary re-renders
+      const parentFieldValue = useWatch({
```

ESLint

[error] React Hook "useWatch" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?

(react-hooks/rules-of-hooks)